target/linux/bcm53xx/patches-5.4/340-ARM-BCM4708-Add-DT-for-Iptime-A5004NS.patch

```diff
@@ -0,0 +1,121 @@
+From 51d1ba0fbf01a7c216e12f10fe0ff8044b012f30 Mon Sep 17 00:00:00 2001
+From: sodo <djdisodo@gmail.com>
+Date: Sun, 14 Mar 2021 17:11:32 +0900
+Subject: [PATCH] ARM: BCM4708: Add DT for Iptime A5004NS
+
+Signed-off-by: Jisoo Lee <djdisodo@gmail.com>
+---
+ arch/arm/boot/dts/Makefile                   |  1 +
+ arch/arm/boot/dts/bcm4708-iptime-a5004ns.dts | 88 ++++++++++++++++++++
+ 2 files changed, 89 insertions(+)
+ create mode 100644 arch/arm/boot/dts/bcm4708-iptime-a5004ns.dts
+
+diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
+index 8e5d4ab4e..650227653 100644
+--- a/arch/arm/boot/dts/Makefile
++++ b/arch/arm/boot/dts/Makefile
+@@ -95,6 +95,7 @@ dtb-$(CONFIG_ARCH_BCM_5301X) += \
+ 	bcm4708-asus-rt-ac56u.dtb \
+ 	bcm4708-asus-rt-ac68u.dtb \
+ 	bcm4708-buffalo-wzr-1750dhp.dtb \
++	bcm4708-iptime-a5004ns.dtb \
+ 	bcm4708-linksys-ea6300-v1.dtb \
+ 	bcm4708-linksys-ea6500-v2.dtb \
+ 	bcm4708-luxul-xap-1510.dtb \
+diff --git a/arch/arm/boot/dts/bcm4708-iptime-a5004ns.dts b/arch/arm/boot/dts/bcm4708-iptime-a5004ns.dts
+new file mode 100644
+index 000000000..91d16bd3e
+--- /dev/null
++++ b/arch/arm/boot/dts/bcm4708-iptime-a5004ns.dts
+@@ -0,0 +1,88 @@
++// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
++/*
++ * Broadcom BCM470X / BCM5301X ARM platform code.
++ * DTS for Iptime a5004ns
++ *
++ * Copyright (C) 2021 Lee Jisoo <djdisodo@gmail.com>
++ */
++
++/dts-v1/;
++
++#include "bcm4708.dtsi"
++
++/ {
++	compatible = "iptime,a5004ns", "brcm,bcm4708";
++	model = "ipTIME A5004NS (BCM4708)";
++
++	chosen {
++                bootargs = "console=ttyS0,115200";
++        };
++
++
++	memory@0 {
++		device_type = "memory";
++		reg = <0x00000000 0x08000000
++		       0x88000000 0x08000000>;
++	};
++
++	leds {
++		compatible = "gpio-leds";
++
++		power {
```

Comment:
Really? The [stock FW](https://github.com/iptime-gpl/userapps_a5004ns/blob/master/fs/bcmac/clones/a5004ns/si/hw#L31) says gpio3 is for wlan2g LED.

Reply:
@mans0n i'll remove that, i guess neither power led nor wlan2g is controllable
and use cpu led as power instead

Reply:
Typically for ipTIME devices, power LED is not controllable, it is directly connected to VCC.
If you rename cpu to power then it will work out of the box but it might confuse users whether it is really a power LED or a CPU LED.